src/fromager/settings.py

```diff
@@ -17,18 +17,58 @@ def pre_built(self, variant: str) -> set[str]:
         names = p.get(variant) or []
         return set(overrides.pkgname_to_override_module(n) for n in names)
 
-    def download_source(self) -> dict[str, dict[str, str]]:
-        return self._data.get("download_source") or {}
-
-    def sdist_download_url(self, pkg: str) -> str | None:
-        p = self.download_source()
-        download_source = p.get(pkg) or {}
-        return download_source.get("url")
-
-    def sdist_local_filename(self, pkg: str) -> str | None:
-        p = self.download_source()
-        download_source = p.get(pkg) or {}
-        return download_source.get("rename_to")
+    def package(self) -> dict[str, dict[str, str]]:
+        return self._return_value_or_default(self._data.get("packages"), {})
+
+    def download_source_url(self, pkg: str, default: str | None = None) -> str | None:
+        download_source = self._get_package_download_source_settings(pkg)
+        return self._return_value_or_default(download_source.get("url"), default)
+
+    def download_source_destination_filename(
+        self, pkg: str, default: str | None = None
+    ) -> str | None:
+        download_source = self._get_package_download_source_settings(pkg)
+        return self._return_value_or_default(
+            download_source.get("destination_filename"), default
+        )
+
+    def resolver_sdist_server_url(
+        self, pkg: str, default: str | None = None
+    ) -> str | None:
+        resolve_dist = self._get_package_resolver_dist_settings(pkg)
+        return self._return_value_or_default(
+            resolve_dist.get("sdist_server_url"), default
+        )
+
+    def resolver_include_wheels(
+        self, pkg: str, default: bool | None = None
+    ) -> bool | None:
+        resolve_dist = self._get_package_resolver_dist_settings(pkg)
+        return self._return_value_or_default(
+            resolve_dist.get("include_wheels"), default
+        )
+
+    def resolver_include_sdist(
+        self, pkg: str, default: bool | None = None
+    ) -> bool | None:
+        resolve_dist = self._get_package_resolver_dist_settings(pkg)
+        return self._return_value_or_default(resolve_dist.get("include_sdist"), default)
```

Comment:
This could be inlined more simply as

```suggestion
        return resolve_dist.get("include_sdist", default)
```

Reply:
An empty section in the `settings.yaml` can break this. For example:
```yaml
packages:
   foo:
       download_source:
            url: "url"
       resolver_dist:
```
The raw data dict for this yaml will have the key `resolver_dist` and its value will be None. When using the in inbuilt `get` method, python returns None instead of the default since the key was defined. We want the default to be returned even when the value is None